### PR TITLE
feat: Add experimental score logic to VisynRanking

### PR DESCRIPTION
**Pull request by puehringer**
Closes *list issues numbers here*

### Developer Checklist (Definition of Done)

**Issue**

- [ ] All acceptance criteria from the issue are met
- [ ] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [ ] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [ ] Branch is up-to-date with the branch to be merged with, i.e., develop
- [ ] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [ ] Descriptive title for this pull request is provided (will be used for release notes later)
- [ ] Reviewer and assignees are defined
- [ ] Add type label (e.g., *bug*, *feature*) to this pull request
- [ ] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [ ] The PR is connected to the corresponding issue (via `Closes #...`)
- [ ] [Summary of changes](#summary-of-changes) is written

### Summary of changes

- 

### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

### CODE REVIEW

File: src/demo/MainApp.tsx [hunk 1]
```diff
@@ -61,8 +84,30 @@ export function MainApp() {
...
+            <Select
+              placeholder="Add a score column"
+              onChange={async (value) => {
+                setLoading(true);
+                // eslint-disable-next-line no-promise-executor-return
+                await new Promise((resolve) => setTimeout(resolve, 1000));
+                createScoreColumnFunc.current(await (value === 'number' ? MyNumberScore(value) : MyStringScore(value)));
+                setLoading(false);
+              }}
+              rightSection={loading ? <i className="fas fa-spinner" /> : undefined}
```
Comment (puehringer) [inline src/demo/MainApp.tsx:98]: Add proper loading spinner.

File: src/demo/MainApp.tsx [hunk 2]
```diff
@@ -61,8 +84,30 @@ export function MainApp() {
       }
     >
       {user ? (
-        <SimpleGrid cols={2} style={{ height: '100%' }}>
-          <VisynRanking data={iris} selection={selection} setSelection={setSelection} />
+        <SimpleGrid cols={2} style={{ height: '100%' }} ml="md" pt="md">
+          <Stack>
+            <Select
```
Comment (puehringer) [inline src/demo/MainApp.tsx:89]: Add "better" UI for adding scores.

File: src/ranking/EagerVisynRanking.tsx [hunk 3]
```diff
@@ -15,6 +16,12 @@ export const defaultBuilder = ({ data }) => {
   return b;
 };
 
+export interface IBuiltVisynRanking {
+  provider: LocalDataProvider;
+  ranking: Ranking;
+  createScoreColumn: (score: IScoreResult) => void;
```
Comment (puehringer) [inline src/ranking/EagerVisynRanking.tsx:22]: Add detailed docstring explaining how to use this feature. Including how to store the ref to the function...

File: src/ranking/score/interfaces.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import castArray from 'lodash/castArray';
+
+export interface ISingleScoreResult {
+  data: unknown[];
+  builder: ColumnBuilder<IValueColumnDesc<unknown>>;
+}
+
+export type IScoreResult = ISingleScoreResult | ISingleScoreResult[];
+
+export async function createScoreColumn(desc: IScoreResult, lineup: LineUp | Taggle, ranking: Ranking): Promise<void> {
```
Comment (puehringer) [inline src/ranking/score/interfaces.ts:11]: Document this function.

File: src/ranking/score/interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import LineUp, { ColumnBuilder, IValueColumnDesc, Ranking, Taggle } from 'lineupjs';
+import castArray from 'lodash/castArray';
+
+export interface ISingleScoreResult {
+  data: unknown[];
```
Comment (puehringer) [inline src/ranking/score/interfaces.ts:5]: Document this interface.

File: src/demo/MainApp.tsx [hunk 6]
```diff
@@ -1,11 +1,32 @@
...
 import { Vis, ESupportedPlotlyVis, ENumericalColorScaleType, EScatterSelectSettings, IVisConfig } from '../vis';
 import { fetchIrisData } from '../vis/stories/Iris.stories';
 import { iris } from '../vis/stories/irisData';
 import { useVisynAppContext, VisynApp, VisynHeader } from '../app';
 import { LoginUtils } from '../security';
 import { VisynRanking } from '../ranking';
+import { IBuiltVisynRanking } from '../ranking/EagerVisynRanking';
+import { IScoreResult } from '../ranking/score';
+
+async function MyStringScore(value: string): Promise<IScoreResult> {
```
Comment (puehringer) [inline src/demo/MainApp.tsx:13]: Move to separate scores file in the demo folder.

File: src/ranking/score/interfaces.ts (hunk 5 above)
Comment (puehringer) [reply src/ranking/score/interfaces.ts:10]: @dvzacharycutler Please don't document the `data` in the interface, but the data property itself. I.e.

```
/**
 * the data ...
 */
data: unknown[];
```

**Review by dvdanielamoitzi — APPROVED** [review]

**Pull request merged**